eul/Store/PreferenceStore.swift

```diff
@@ -98,6 +103,21 @@ class PreferenceStore: ObservableObject {
         availableComponents.remove(at: index)
     }
 
+    func toggleActiveMenuComponent(at index: Int) {
+        isActiveMenuComponentToggling = true
+        availableMenuComponents.append(activeMenuComponents[index])
+        activeMenuComponents.remove(at: index)
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) { // wait for rendering, will crash w/o delay
+            self.isActiveMenuComponentToggling = false
+        }
+    }
```

Comment:
noticed we have the same pattern for status bar components and menu components, it's ok to copy the logic for now since we'll add more menu components soon. leave this comment for tracking